### DISCUSSION TITLE
Computing orbit forecasts/predictions

### DISCUSSION THREAD

**Discussion by Yang-Hu-LNTU** [groops-devs/groops#122] [Q&A]
I want to fit the precise orbital period 00-22:45 (with stochasticPlus), and then use SimulateOrbit to forecast the 23:00-23:45 orbits, and obtain the millimeter level forecast orbits.
This is my current setting. Forces is the same as the previous setting.

Only centimeter level orbits can be obtained by polynomial fitting. 
The problem is that when SimulateOrbit is used, it can only predict 15 minutes. When the prediction duration increases, it will prompt that the input file is missing data. How do I need to set it to achieve the prediction accuracy and achieve millimeter level accuracy? Such as timeSeries, integrationConstants or propagator.

**Comment by akvas** [comment]
Hi @Yang-Hu-LNTU,

a very similar task is given in our [GNSS network cookbook](https://groops-devs.github.io/groops/html/cookbook.gnssNetwork.html) in the Preprocessing: Orbit section.

The key points are to use [InstrumentResample](https://groops-devs.github.io/groops/html/InstrumentResample.html) to extrapolate the initial orbit to the full 24 hours (set maxDataPointRange and maxExtrapolationDistance accordingly) and then use [PreprocessingVariationalEquation](https://groops-devs.github.io/groops/html/PreprocessingVariationalEquation.html) and [PreprocessingVariationalEquationOrbitFit](https://groops-devs.github.io/groops/html/PreprocessingVariationalEquationOrbitFit.html) to fit the 24h orbit to your 00:00 - 22:45 observation period.

Cheers,
Andreas

**Reply by Yang-Hu-LNTU** [reply]
Thank you for your guidance. I have realized the above scheme now.
I also have a question about how to extrapolate forward (such as 23:00-00 the day before)? It hasn't been resolved yet。
Best wishes.

**Reply by Yang-Hu-LNTU** [reply]
Thank you for your guidance. I also have another question similar to that.
I have a precise orbit products from 02:00 to 22:00, now i want to abtain 24h orbit.

All sets like:
--InstrumentResample: 
       polynomialDegree: 7
       maxDataPointRange: 7200
       maxExtrapolationDistance: 7200
       timeSeries from 00:00 to 23:45, sampling 30s
--PreprocessingVariationalEquation:
       can get 24h output Variational file
--PreprocessingVariationalEquationOrbitFit
       inputOrbit : 02:00-22:00, sampling 15min
       inputVariational: 00:00-23:45, sampling 30s

But errors occurred when accumulate normal equations:
       tips like   B := N(232 x 232)^-1 * B = (232 x 1)
                      W'W = A(232 x 232)
                      cannot compute decomposition, error = 227
If instrumentResample time set like 02:00-23:45, all program can pass on and i can get precise orbit from 02:00-23:45. I think the problem is in prediction for the time 00:00 to 02:00. 

How do I need to set it up, or is this feature not currently available。

Looking forward to your answer。
Best wishes.

**Comment by akvas** [comment]
Hi @Yang-Hu-LNTU,

GROOPS does not yet feature a straightforward way to predict orbits from scratch. You would need some approximate orbit which is given over the whole 24 hours. This can be a simple Keplerian orbit from TLEs, broadcast orbits, or similar for the first computation of the variational equations. Hope this helps!

Cheers,
Andreas
